torch/_inductor/utils.py

```diff
@@ -856,6 +856,59 @@ def argsort(seq) -> List[int]:
     return list(reversed(sorted(a_r, key=getter, reverse=True)))  # noqa: C413
 
 
+def argsort_sym(seq: List[Union[int, torch.SymInt, sympy.Expr]]) -> List[int]:
+    # Strategy: convert all symints to sympy.Expr, then evaluate them.
+    seq_with_exprs = []
+    maybe_shape_env = None
+    for idx, maybe_symint in enumerate(seq):
```

Comment:
why do we need to do this ? inductor should always have a shape env ?

Reply:
> Looks good - just had one question about where you are getting the non shape env case. I dont want to leave misleading impressions about inductor invariants in code

The problem is get_stride_order is called from both inside and outside of inductor code. Would you prefer me to create two versions of get_stride_order -- one for calling inside of inductor, and one for calling outside of inductor?

The only callsite not in inductor is [here](https://www.internalfb.com/code/fbsource/[662193f8e44289b217d421f47ff9b96573060081]/fbcode/caffe2/torch/_higher_order_ops/flex_attention.py?lines=63), so it's not too bad to duplicate the code.

Reply:
A couple options:

a) The callsite there can probably use this same logic from prims - would prefer that for not mixing up the symint/sympy expr :

```
    logical_to_physical_perm = compute_elementwise_output_logical_to_physical_perm(
        *tensors, _skip_checks=True
    )
    permuted_shape = apply_perm(shape, logical_to_physical_perm)  # to physical
```

b) Can we have a common api where we take in the shape env ? in inductor, should always be V.graph.sizevars.shape_env ? whereas in tracing should be in fake mode.

Reply:
Updated to have an API where we accept the shape_env